test/common/http/filter/ratelimit_test.cc

```diff
@@ -215,5 +233,94 @@ TEST_F(HttpRateLimitFilterTest, ResetDuringCall) {
   filter_callbacks_.reset_callback_();
 }
 
+class HttpRateLimitFilterRequestHeadersTest : public testing::Test {
```

Comment:
Lot of duplication here. Either use a base class for common stuff, or just refactor the main test fixture so that you pass the JSON configuration via a setup() function